core/src/test/scala/unit/kafka/cluster/PartitionTest.scala

```diff
@@ -412,14 +506,26 @@ class PartitionTest extends AbstractPartitionTest {
     def assertReadRecordsError(error: Errors,
                                currentLeaderEpochOpt: Optional[Integer]): Unit = {
       try {
-        partition.readRecords(
-          lastFetchedEpoch = Optional.empty(),
-          fetchOffset = 0L,
-          currentLeaderEpoch = currentLeaderEpochOpt,
+        val fetchOffset = 0L
+        val logStartOffset = 0L
+
+        val fetchPartitionData = new FetchRequest.PartitionData(
+          Uuid.ZERO_UUID,
+          fetchOffset,
+          logStartOffset,
+          Int.MaxValue,
+          currentLeaderEpochOpt,
+          Optional.empty[Integer]
+        )
+
+        partition.fetchRecords(
+          fetchParams = followerFetchParams(remoteReplicaId),
+          fetchPartitionData = fetchPartitionData,
+          fetchTimeMs = time.milliseconds(),
           maxBytes = 1024,
-          fetchIsolation = FetchLogEnd,
-          fetchOnlyFromLeader = true,
-          minOneMessage = false)
+          minOneMessage = false,
+          updateFetchState = true
+        )
         if (error != Errors.NONE)
           fail(s"Expected readRecords to fail with error $error")
       } catch {
```

Comment:
Not related to your change but it seems that we should have used `assertThrows` here.